src/services/DexPoolScanner.ts

```diff
@@ -0,0 +1,180 @@
+import {injectable} from 'inversify';
+import {Logger} from 'winston';
+import {StaticJsonRpcProvider} from '@ethersproject/providers';
+
+import {ChainsIds} from '../types/ChainsIds.js';
+import {DexProtocolName} from '../types/DexProtocolName.js';
+import {MappingRepository} from '../repositories/MappingRepository.js';
+import {BlockchainProviderRepository} from '../repositories/BlockchainProviderRepository.js';
+import {DexProtocolFactory, PoolCreatedEvent} from '../factories/DexProtocolFactory.js';
+import {DexProtocolInterface} from '../interfaces/DexProtocolInterface.js';
+import {DexRepository} from '../repositories/DexRepository.js';
+import Application from '../lib/Application.js';
+import Settings from '../types/Settings.js';
+
+export type Result = {
+  fromBlock: number;
+  toBlock?: number;
+  success: boolean;
+  synchronized?: boolean;
+  waitBlockRange?: boolean;
+};
+
+export type UniswapPoolCreatedEvent = {
+  token0: string;
+  token1: string;
+  fee: bigint;
+  pool: string;
+  anchor: number;
+};
+
+@injectable()
+export class DexPoolScanner {
+  logger: Logger;
+  settings: Settings;
+  mappingRepository!: MappingRepository;
+  blockchainProviderRepository!: BlockchainProviderRepository;
+  dexRepository!: DexRepository;
+
+  startBlock: number;
+  dexProtocolName!: DexProtocolName;
+  step: number;
+  provider: StaticJsonRpcProvider | undefined;
+  chainId!: ChainsIds;
+  dexProtocol: DexProtocolInterface;
+  logPrefix: string;
+
+  constructor(chainId: ChainsIds, dexProtocolName: DexProtocolName) {
+    this.blockchainProviderRepository = Application.get(BlockchainProviderRepository);
+    this.mappingRepository = Application.get(MappingRepository);
+    this.dexRepository = Application.get(DexRepository);
+    this.provider = <StaticJsonRpcProvider>this.blockchainProviderRepository.get(chainId);
+    this.dexProtocol = DexProtocolFactory.create(chainId, dexProtocolName, this.provider);
+    this.settings = Application.get('Settings');
+    this.logger = Application.get('Logger');
+    this.step = 0;
+    this.logPrefix = `[DexPoolScanner][${chainId}][${dexProtocolName}]`;
+
+    if (!chainId || !dexProtocolName) {
+      throw new Error(`${this.logPrefix} chainId or dexProtocolName not set`);
+    }
+
+    this.chainId = chainId;
+    this.dexProtocolName = dexProtocolName;
+
+    if (!this.settings.blockchains[chainId].providerUrl.join('')) {
+      throw new Error(`${this.logPrefix} provider URL not set`);
+    }
+
+    const startBlock = this.settings.dexes[chainId]?.[dexProtocolName]?.startBlock;
+
+    if (!startBlock || Number(startBlock) <= 0) {
+      throw new Error(`${this.logPrefix} startBlock must be higher than 0`);
+    }
+
+    this.startBlock = startBlock!;
+    this.step = this.settings.dexes[chainId]?.[dexProtocolName]?.agentStep || 0;
+  }
+
+  async run(): Promise<Result> {
+    const fromBlock = await this.getMarker();
```

Comment:
please use mapping model to store last sync blocks, I think we can remove marker model